### DISCUSSION TITLE
Multiple proxy servers with different authentication methods.

### DISCUSSION THREAD

**Discussion by davibicudo** [genotrance/px#201] [Q&A]
Hi!

First of all, I'm not a network security expert, just someone suffering from acute proxy frustration (kind of an epidemic in my team).

My company provides two different proxy servers, one with authentication and another without it. The authentication for the former is username+password, but not the same as my Windows credentials. 
We usually set proxy via the two http/s_proxy environment variables. 
Sometimes things work only without a proxy at all, sometimes with one, sometimes with another. Very unclear. 
I believe px can help us, since it tries one server after another until things "work" (does it also try without a proxy at all?). But looking at the documentation, I couldn't find a clear way to solve the varying authentication methods. 
Advice on this would be much appreciated!

Best,
davibicudo

**Comment by genotrance** [comment]
If no proxy is configured, Px will connect directly. However, if a proxy is connected, it won't try that.

Can you try configuring Px with both proxies and logging enabled and running some tests? It will help to know what's happening with both proxies and which use cases fail.
